### ISSUE TITLE
Option to enable/disable IPv6

### ISSUE THREAD

**Issue by orlandothoeny** [php-amqplib/php-amqplib#1169]
**Description**  
We cannot use the library since version 3.6.1, because of https://github.com/php-amqplib/php-amqplib/pull/1145.

We use https://github.com/cloudamqp/amqproxy which only listens on IPv4. Our application connects to this proxy via `localhost`, which has the default `::1     localhost ip6-localhost ip6-loopback` entry in the server's `/etc/hosts` file.

Possible fixes:
* Either make it that IPv6 support must be explicitly enabled or add an option that allows disabling IPv6
* Have some sort of check that validates that IPv6 is actually working, not sure though if that would be too expensive performance-wise

**Example**  
We get the following error:
```
In SocketIO.php line 82:
                                                        
  [PhpAmqpLib\Exception\AMQPIOException (111)]          
  Error Connecting to server (111): Connection refused
```

**Comment by ramunasd** [comment]
Just use `127.0.0.1` for connection if proxy is listening on local loopback.

**Comment by orlandothoeny** [comment]
We're using the default `/etc/hosts` of Ubuntu.

Yes, changing to an IP (`127.0.0.1`) would probably be the easiest fix.

**Comment by lukebakken** [comment]
Hmm, I'm surprised the ipv6 address is picked up first.

```
$ docker run --pull always --tty --interactive ubuntu:latest
latest: Pulling from library/ubuntu
bccd10f490ab: Already exists
Digest: sha256:77906da86b60585ce12215807090eb327e7386c8fafb5402369e421f44eff17e
Status: Downloaded newer image for ubuntu:latest
root@0e7320e07973:/# cat /etc/hosts
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
172.17.0.2      0e7320e07973
```

**Comment by orlandothoeny** [comment]
Yes it's strange, the `127.0.0.1       localhost` record is also the first one on our servers

**Comment by orlandothoeny** [comment]
Oh no, wait a second..
That was my local one.

Seems like this changed in some Ubuntu version.
That's the actual one from the server:
```
127.0.0.1	localhost

# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
```

**Comment by denis-sychev28** [comment]
Hi, i get error "Error Connecting to server (-10001): Unknown host" in PhpAmqpLib/Wire/IO/SocketIO.php:82, if run in ubuntu 22, function checkdnsrr in PhpAmqpLib/Wire/IO/SocketIO.php:321 return false for m-translate.com and return true if run in docker container fedora
